### PR TITLE
Brocade VDX switch integration as part of switch discovery

### PR DESCRIPTION
**Pull request by dpasupathi**
Added support for brocade switch discovery. Following changes are made

- taskrunner.py:
      Modified the script to replace the functions which is not compatible with python 3.
- data/templates/brocade-catalog-version.py:
      script used to catalog as part of auto discovery process
- data/templates/brocade-vdx-deploy-config.py:
      script used to deploy a config file to running

### CODE REVIEW

**Comment by benbp** [comment]
FYI @pscharla @johren

**Comment by benbp** [comment]
@dpasupathi I assume the answer is yes, but taskrunner.py will work for _both_ python 2 and 3 now?

**Comment by dpasupathi** [comment]
@benbp: It should work. @pscharla ran it against Arista and it seems to work. But i haven't tested it on python 2. Can you please help me to test against cisco box?

**Comment by yyscamper** [comment]
jenkins: ok to test

**Comment by lanchongyizu** [comment]
Can one of the admins verify this patch?

**Comment by jlongever** [comment]
test this please

**Comment by dpasupathi** [comment]
I had tested this on my environment. But i would like some to test on python 2.x environment to make sure nothing is broken.

**Comment by lanchongyizu** [comment]
Can one of the admins verify this patch?

**Comment by benbp** [comment]
👍

File: data/profiles/taskrunner.py [hunk 1]
```diff
@@ -57,26 +67,35 @@ def download_script(downloadUrl):
...
+        session.close()
 
     for _ in range(3):
         try:
-            print "Posting task data\n {}".format(task_data)
-            urllib2.urlopen(req)
-        except urllib2.URLError:
+            print("Posting task data\n {}".format(task_data))
+            session = requests.Session()
+            response = requests.Response()
```
Comment (benbp) [inline data/profiles/taskrunner.py]: Is this necessary? `response` just gets redefined the line below?

**Comment by dpasupathi** [comment]
@benbp : Removed the unnecessary response initialization. Please review.

**Comment by dpasupathi** [comment]
@JenkinsRHD @benbp : Is this error because of my new changes atleast this worked in my test environment.

**Comment by benbp** [comment]
👍

**Comment by benbp** [comment]
test this please

**Comment by benbp** [comment]
@dpasupathi no I think it's a one-off error, you can ignore it (I re-kicked the tests).

**Comment by brianparry** [comment]
test this please

**Comment by anhou** [comment]
:+1:

**Pull request merged**